website/content/docs/agent/index.mdx

```diff
@@ -92,6 +116,218 @@ When running under `systemd` on Linux, Consul notifies systemd by sending
+| `retry_join` | Array of string values that specify one or more agent addresses to join after startup. The agent will continue trying to join the specified agents until it has successfully joined another member. <br/>See [`-retry-join`](/docs/agent/options#_retry_join) for details. | none                                                                                                             |
+| `addresses`  | Block of nested objects that define addresses bound to the agent for internal cluster communication.                                                                                                                                                                       | `"http": "0.0.0.0"` See the Agent Configuration page for [default address values](/docs/agent/options#addresses) |
+| `ports`      | Block of nested objects that define ports bound to agent addresses. <br/>See (link to addresses option) for details.                                                                                                                                                       | See the Agent Configuration page for [default port values](/docs/agent/options#ports)                            |
+
+### Server Node in a Service Mesh
+
+The following example configuration is for a server agent named "`consul-server`". The server is [bootstrapped](/docs/agent/options#_bootstrap) and the Consul GUI is enabled.
+The reason this server agent is configured for a service mesh is that the `connect` configuration is enabled. Connect is Consul's service mesh component that provides service-to-service connection authorization and encryption using mutual Transport Layer Security (TLS). Applications can use sidecar proxies in a service mesh configuration to establish TLS connections for inbound and outbound connections without being aware of Connect at all. See [Connect](/docs/connect) for details.
+
+<Tabs>
+<Tab heading="HCL">
+
+```hcl
+
+node_name = "consul-server"
+server = true
+bootstrap = true
+ui_config {
+        enabled = true
+    }
+datacenter = "dc1"
+data_dir = "/consul/data"
+log_level = "INFO"
+addresses {
+        http = "0.0.0.0"
+    }
+connect  {
+        enabled = true
+    }
+
+```
+
+</Tab>
+<Tab heading="JSON">
+
+```json
+{
+  "node_name": "consul-server",
+  "server": true,
+  "bootstrap": true,
+  "ui_config": {
+    "enabled": true
+  },
+  "datacenter": "dc1",
+  "data_dir": "consul/data",
+  "log_level": "INFO",
+  "addresses": {
+    "http": "0.0.0.0"
+  },
+  "connect": {
+    "enabled": true
+  }
+}
+```
+
+</Tab>
+</Tabs>
+
+### Client Node Registering a Service
```

Comment:
The order feels a bit odd. We went from server back to client context. And then after this section we jump back to server context. Maybe we should move this after we wrap up the server discussion ?